test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -498,6 +676,213 @@ def forward(self, x):
+        self.run_test(model, other_input, batch_size=RNN_BATCH_SIZE + 1)
+
+
+def make_test(name, base, layer, bidirectional, initial_state,
+              variable_length, dropout,
+              **extra_kwargs):
+    test_name = str('_'.join([
+        'test', name, layer[1],
+        bidirectional[1], initial_state[1],
+        variable_length[1], dropout[1]
+    ]))
+
+    # Cannot export with older opsets because of 'ConstantFill' op
+    # ConstantFill was a temp op removed at opset 8. This is no longer supported by onnxruntime
+    @skipIfUnsupportedMinOpsetVersion(9)
+    def f(self):
+        self._dispatch_rnn_test(
+            base,
+            layers=layer[0],
+            bidirectional=bidirectional[0],
+            initial_state=initial_state[0],
+            packed_sequence=variable_length[0],
+            dropout=dropout[0],
+            **extra_kwargs)
+
+    f.__name__ = test_name
+    setattr(TestONNXRuntime, f.__name__, f)
+
+
+def setup_rnn_tests():
+    layers_opts = [
+        (1, 'unilayer'),
+        (3, 'trilayer')
+    ]
+    bidirectional_opts = [
+        (False, 'forward'),
+        (True, 'bidirectional')
+    ]
+    initial_state_opts = [
+        (True, 'with_initial_state'),
+        (False, 'no_initial_state')
+    ]
+    variable_length_opts = [
+        (0, 'without_sequence_lengths'),
+        (1, 'with_variable_length_sequences'),
+        (2, 'with_batch_first_sequence_lengths')
+    ]
+    dropout_opts = [
+        (0.2, 'with_dropout'),
+        (0.0, 'without_dropout')
+    ]
+    test_count = 0
+    for (layer, bidirectional, initial_state, variable_length, dropout) in \
+        itertools.product(
+            layers_opts,
+            bidirectional_opts,
+            initial_state_opts,
+            variable_length_opts,
+            dropout_opts,
+    ):
+
+        for base, name, extra_kwargs in (
+                ('elman', 'elman_relu', {'nonlinearity': u'relu'}),
+                ('elman', 'elman_tanh', {'nonlinearity': u'tanh'}),
+                ('lstm', 'lstm', {}),
+                ('gru', 'gru', {})
+        ):
+            # This is a hack to skip elman_rnn bidirectional tests for now
+            # TODO: Revert this once elman_rnn bidirectional issue is fixed
+            if base == 'elman' and bidirectional[1] == 'bidirectional':
+                continue
+            make_test(name, base, layer, bidirectional, initial_state,
+                      variable_length, dropout,
+                      **extra_kwargs)
+            test_count += 1
+
+    # sanity check that a representative example does exist
+    TestONNXRuntime.test_gru_trilayer_forward_with_initial_state_without_sequence_lengths_with_dropout
+
+    # make sure no one accidentally disables all the tests without
+    # noticing
+    # assert test_count == 192, test_count
+    # TODO: Revert this once elman_rnn bidirectional issue is fixed
+    assert test_count == 144, test_count
```

Comment:
better to check and raise here, some linters complain that `assert` can be disabled.